source/win32_tinyengine.c

```diff
@@ -163,6 +391,819 @@ Win32MainWindowCallback(HWND Window, UINT Message, WPARAM WParam, LPARAM LParam)
             IsRunning = false;
         } break;
 
+        case WM_INPUT:
+        {
+            UINT SizeOfBuffer = sizeof(RAWINPUT);
+            char Data[sizeof(RAWINPUT)] = {0};
+            if(GetRawInputData((HRAWINPUT)LParam, RID_INPUT, Data, &SizeOfBuffer, sizeof(RAWINPUTHEADER)) <= SizeOfBuffer)
```

Comment:
Isn't the point of getting the size to use it for this Data variable? If i remember the docs say call GetRawInputData with the Data var being null, then you call it again with a buffer that is the size that you got back from  GetRawInputData

Reply:
Oops, yes this is true. Originally I had a TODO here, but I think I got rid of it. It was asking whether or not we wanted to dynamically allocate here or not. Then I noticed that on the molecular-matters.com post you linked on Discord, it had:

```cpp
char buffer[sizeof(RAWINPUT)] = {};
UINT size = sizeof(RAWINPUT);
GetRawInputData(reinterpret_cast<HRAWINPUT>(lParam), RID_INPUT, buffer, &size, sizeof(RAWINPUTHEADER));
```

Do you think that's ok, or should I do it the MDSN recommended way? Also, my code should probably say `Data[SizeOfBuffer]` since I switched the order

Reply:
I mean if it works and we don't have to do a dynamic allocation thats a win i guess?

Go Hayden.